src/digital/m2kdigital_impl.cpp

```diff
@@ -493,3 +493,38 @@ void M2kDigitalImpl::resetRateMux()
 {
 	m_dev_read->setStringValue("rate_mux", "logic_analyzer");
 }
+
+void M2kDigitalImpl::setExternalClocksource(bool external)
+{
+	if (m_dev_generic->hasGlobalAttribute("clocksource")) {
+		if (external) {
+			m_dev_generic->setStringValue("clocksource", "external");
+		} else {
+			m_dev_generic->setStringValue("clocksource", "internal");
+		}
+	} else {
+		THROW_M2K_EXCEPTION("M2kDigital: "
+				    "The clock source cannot be configure on "
+				    "the current board; Check the firmware version.",
+				    libm2k::EXC_INVALID_FIRMWARE_VERSION);
+	}
+
+}
+
+bool M2kDigitalImpl::isClocksourceExternal()
+{
+	if (m_dev_generic->hasGlobalAttribute("clocksource")) {
+		if (m_dev_generic->getStringValue("clocksource") == "external") {
+			return true;
+		} else {
+			return false;
+		}
+	} else {
+		THROW_M2K_EXCEPTION("M2kDigital: "
+				    "There is no clocksource attribute on "
+				    "the current board; Check the firmware version.",
+				    libm2k::EXC_INVALID_FIRMWARE_VERSION);
+	}
```

Comment:
Add return false after throw, because in some cases THROW_M2K_EXCEPTION can have exceptions disabled.